File: android/src/main/kotlin/io/agora/agora_rtc_engine/ResultCallback.kt
package io.agora.agora_rtc_engine

import io.agora.rtc.base.Callback
import io.flutter.plugin.common.MethodChannel.Result

class ResultCallback(
  private val result: Result?
) : Callback() {
  override fun success(data: Any?) {
    result?.success(data)
  }

  override fun failure(code: String, message: String) {
    result?.error(code, message, null)
  }
}

File: ios/Classes/ResultCallback.swift
import Foundation
import AgoraRtcKit

class ResultCallback: NSObject, Callback {
    private var result: FlutterResult?

    init(_ result: FlutterResult? = nil) {
        self.result = result
    }

    func success(_ data: Any?) {
        result?(data)
    }

    func failure(_ code: String, _ message: String) {
        result?(FlutterError.init(code: code, message: message, details: nil))
    }
}
